rq/worker.py

```diff
@@ -677,6 +678,10 @@ def handle_job_failure(self, job: 'Job', queue: 'Queue', started_job_registry=No
             else:
                 enqueue_dependents = True
 
+            if job.group_id:
+                group = Group.fetch(job.group_id, self.connection)
+                group.cleanup(pipeline=pipeline)
```

Comment:
Having `group.cleanup()` convenience method is nice, but we should also have a class method that accepts Redis connection and `group_id`. In this context, `Group.fetch()` incurs an extra Redis call which can be avoided.

Reply:
Moved the logic to classmethod, let me know if that looks ok